modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2067,17 +2067,23 @@ def sort_function(df):
         if sampling_probability >= 1:
             from modin.config import MinPartitionSize
 
-            if (
-                len(self.index) < MinPartitionSize.get()
-                and self._partitions.shape[1] == 1
-            ):
+            ideal_num_new_partitions = round(len(self.index) / MinPartitionSize.get())
+            if len(self.index) < MinPartitionSize.get() or ideal_num_new_partitions < 2:
+                if self._partitions.shape[1] != 1:
+                    # In this case, we have more than one column partition, so we first need
+                    # to create row-wise partitions with all of the columns, so we don't have
+                    # a KeyError when sorting. This method can be slow if we have very very
+                    # many columns.
+                    new_partitions = self._partition_mgr_cls.row_partitions(
+                        self._partitions
+                    )
+                    self._partitions = np.array(
+                        [[partition] for partition in new_partitions]
+                    )
```

Comment:
I'm slightly wary of changing `self` here, shouldn't our objects be immutable?

Reply:
Makes sense - I've gone ahead and made it so we create a new dataframe object with the new partitions!